lesson09/script.js

```diff
@@ -0,0 +1,46 @@
+function showLoadingImg(){
+    const loadingImg = document.createElement('img');
+    loadingImg.src = "loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function removeLoadingImg(){
+    document.getElementById('js-loading').remove();
+}
+
+function showCategories(result){
+    const fragment = document.createDocumentFragment();
+    for(const list of result){
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+        a.textContent = list.text;
+        a.href = `/${list.to}`;
+        img.src = list.img;
+        img.alt = list.alt;
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+    document.getElementById('js-list').appendChild(fragment);
+}
+
+function SetCategoriesAfter3Seconds(){
+    return new Promise(resolve => {
```

Comment:
```suggestion
function SetCategoriesAfter3Seconds(){
    showLoadingImg();
    return new Promise(resolve => {
```
I'm going to use loading before I activate a promise, so how about putting it in this function.
What do you think?

Reply:
"showLoadingImg ()" is not wait 3 seconds.
So I think that the function name and the processing content would not match if I put it in "SetCategoriesAfter3Seconds ()".

Reply:
I understand!

It will be displayed in 3 seconds until the promise is resolved, so I thought it would be okay to implement it together, so I commented like that🙇‍♀

Reply:
Actually I agree with Sen-san's idea.
I think it would be good to add a loading image here since the loading image is displayed while waiting 3 seconds for the promise.

Reply:
@csakyo 
Thank you for your comment!
After listening to your opinion, I thought it would be better to put "showLoadingImg ()" in "SetCategoriesAfter3Seconds ()".

Reply:
@seeeen044 @csakyo 
Thank you for your advice!
I will change to Sen-san's idea.